app/javascript/mastodon/rtl.js

```diff
@@ -23,5 +23,14 @@ export function isRtl(text) {
     return false;
   }
 
+  // Remove mentions before counting characters to decide RTL ratio
+  text = text.replace(/@[0-9A-Za-z_@]+/g, '');
```

Comment:
Instead of all of this it's probably better to just rely on what the language detected in the toot is? And then RTL if it's [arabic, farsi, hebrew, etc]

Then you wouldn't need to duplicate the logic in both ruby and JS